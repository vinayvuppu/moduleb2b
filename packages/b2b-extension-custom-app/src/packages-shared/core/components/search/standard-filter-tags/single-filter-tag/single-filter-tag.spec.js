import React from 'react';
import { shallow } from 'enzyme';
import SingleFilterTag from './single-filter-tag';

const createTestProps = custom => ({
  fieldLabel: 'Date',
  filterTypeLabel: 'Equals to',
  value: '123',
  renderValue: jest.fn(x => x),
  onRemove: jest.fn(),
  onClick: jest.fn(),
  ...custom,
});

describe('rendering', () => {
  it('outputs correct tree', () => {
    const props = createTestProps();
    const wrapper = shallow(<SingleFilterTag {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('callbacks', () => {
  it('renderValue', () => {
    const props = createTestProps();
    shallow(<SingleFilterTag {...props} />);
    expect(props.renderValue).toHaveBeenCalledTimes(1);
    expect(props.renderValue).toHaveBeenLastCalledWith(props.value);
  });
});
