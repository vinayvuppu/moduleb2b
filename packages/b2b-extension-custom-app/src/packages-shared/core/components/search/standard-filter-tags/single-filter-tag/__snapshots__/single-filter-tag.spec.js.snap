// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<Tag
  horizontalConstraint="scale"
  isDisabled={false}
  onClick={[MockFunction]}
  onRemove={[MockFunction]}
  styles={
    Object {
      "body": Object {},
    }
  }
  type="normal"
>
  <div
    className="tagBody"
  >
    <div
      className="bold"
    >
      Date
      :
    </div>
    <div>
      Equals to
    </div>
    <div
      className="bold"
    >
      123
    </div>
  </div>
</Tag>
`;
